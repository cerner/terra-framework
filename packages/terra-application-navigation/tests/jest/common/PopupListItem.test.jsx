import React from 'react';
import PopupMenuListItem from '../../../src/common/_PopupMenuListItem';

describe('PopupMenuListItem', () => {
  it('should render with required prop data', () => {
    const shallowComponent = shallow(
      <PopupMenuListItem
        text="my text"
        onSelect={jest.fn()}
      />,
    );

    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render prop data', () => {
    const shallowComponent = shallow(
      <PopupMenuListItem
        icon={<div>test</div>}
        notificationCount={1}
        text="my text"
        onSelect={jest.fn()}
      />,
    );

    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render with showSelections', () => {
    const shallowComponent = shallow(
      <PopupMenuListItem
        text="my text"
        onSelect={jest.fn()}
        isSelected
        showSelections
      />,
    );

    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render without showSelections', () => {
    const shallowComponent = shallow(
      <PopupMenuListItem
        text="my text"
        onSelect={jest.fn()}
        isSelected
      />,
    );

    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render with selection callback', () => {
    const mockCallBack = jest.fn();
    const mockCallBack2 = jest.fn();
    const mockCallBack3 = jest.fn();

    const shallowComponent = shallow(
      <PopupMenuListItem
        text="my text"
        onSelect={mockCallBack}
      />,
    );

    expect(shallowComponent).toMatchSnapshot();
    shallowComponent.find('.item').simulate('click');
    shallowComponent.find('.item').simulate('keydown', { nativeEvent: { keyCode: 13 }, preventDefault: mockCallBack2, stopPropagation: mockCallBack3 });
    shallowComponent.find('.item').simulate('keydown', { nativeEvent: { keyCode: 32 }, preventDefault: mockCallBack2, stopPropagation: mockCallBack3 });
    expect(mockCallBack.mock.calls.length).toEqual(3);
    expect(mockCallBack2.mock.calls.length).toEqual(2);
    expect(mockCallBack3.mock.calls.length).toEqual(2);
  });
});
