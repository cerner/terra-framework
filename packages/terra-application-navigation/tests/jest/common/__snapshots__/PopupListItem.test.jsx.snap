// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupMenuListItem should render prop data 1`] = `
<li
  className="item"
  data-focus-styles-enabled={true}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="menuitemcheckbox"
  tabIndex="0"
>
  <div
    className="icon"
  >
    <div>
      test
    </div>
  </div>
  <div
    className="text"
  >
    my text
  </div>
  <PopupCount
    className="extension-row-count"
    isHidden={false}
    isInline={true}
    value={1}
  />
</li>
`;

exports[`PopupMenuListItem should render with required prop data 1`] = `
<li
  className="item"
  data-focus-styles-enabled={true}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="menuitemcheckbox"
  tabIndex="0"
>
  <div
    className="text"
  >
    my text
  </div>
  <PopupCount
    className="extension-row-count"
    isHidden={true}
    isInline={true}
    value={0}
  />
</li>
`;

exports[`PopupMenuListItem should render with selection callback 1`] = `
<li
  className="item"
  data-focus-styles-enabled={true}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="menuitemcheckbox"
  tabIndex="0"
>
  <div
    className="text"
  >
    my text
  </div>
  <PopupCount
    className="extension-row-count"
    isHidden={true}
    isInline={true}
    value={0}
  />
</li>
`;

exports[`PopupMenuListItem should render with showSelections 1`] = `
<li
  className="item"
  data-focus-styles-enabled={true}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="menuitemcheckbox"
  tabIndex="0"
>
  <div
    className="selection-icon"
  >
    <IconCheckmark
      data-name="Layer 1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
  <div
    className="text"
  >
    my text
  </div>
  <PopupCount
    className="extension-row-count"
    isHidden={true}
    isInline={true}
    value={0}
  />
</li>
`;

exports[`PopupMenuListItem should render without showSelections 1`] = `
<li
  className="item"
  data-focus-styles-enabled={true}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="menuitemcheckbox"
  tabIndex="0"
>
  <div
    className="text"
  >
    my text
  </div>
  <PopupCount
    className="extension-row-count"
    isHidden={true}
    isInline={true}
    value={0}
  />
</li>
`;
